Encoder

strange-square opened the pull request:
Encoder

strange-square committed:
Encoder

salamantos commented on caesar.py:
```diff
@@ -0,0 +1,31 @@
+        if i.isalpha() and i.islower():
+
+            new_num = alphabet_low[(
+                alphabet_low.find(i) + key) % alphabet_size]
```
find() достаточно дорогая операция, поэтому лучше не использовать ее на каждом символе из текста

лучше сделать предобработку: построить дикт значение-позиция и использовать его. Кстати, поскольку ты используешь это в каждом шифре, стоит вынести логику предобработки в отдельный класс

salamantos commented on caesar.py:
```diff
@@ -0,0 +1,31 @@
+
+
+def caesar_decode(key, text):
+    return caesar_encode(-key, text)
```
не хватает пустой строки в конце файла)

salamantos replied:
проверь остальные файлы тоже

salamantos commented on encryptor.py:
```diff
@@ -0,0 +1,136 @@
+
+    if args.cipher == 'caesar':
+        if not args.key.isdigit():
+            print("Key is not a number")
```
давай решим какой тип кавычек использовать: одинарные или двойные и ограничимся им во всем проекте

salamantos commented on caesar.py:
```diff
@@ -0,0 +1,31 @@
+            res.append(new_num)
+
+        elif i.isalpha() and i.isupper():
+
```
отступы внутри блоков if, for, elif, else делать не обязательно, стоит только если это повышает читаемость кода

salamantos commented on encryptor.py:
```diff
@@ -0,0 +1,136 @@
+        res = vigenere_encode(args.key, text)
+    
+    elif args.cipher == 'vernam':
+        if not args.key.isdigit() or (int(args.key) < 0 or int(args.key) > 25):
```
можно заменить на `not 0 <= int(args.key) <= 25`

salamantos commented on encryptor.py:
```diff
@@ -0,0 +1,136 @@
+    decoder.add_argument('--output-file', dest='output_f',
+                         type=argparse.FileType('w'))
+
+    trainer = subparsers.add_parser('train')
```
много дублирующегося кода, давай попробуем избавиться от этого

salamantos commented on README.md:
```diff
@@ -1 +1,32 @@
+Запуск: 
+
+Шифрование:
+./encryptor.py encode --cipher {caesar|vigenere|vernam} --key {<number>|<word>} [--input-file input.txt] [--output-file output.txt]
```
если запускать без --input-file input.txt --output-file output.txt, то программа почему-то зависает

salamantos reviewed (changes requested):
всё супер, но есть несколько замечаний

strange-square committed:
- В файл README добавлены рекомендации по запуску для исправления
  зависаний
- Предобратка алфавита вынесена в отдельный модуль. Добавлен словарь
  для ускорения поиска по алфавиту
- Удалены отступы внутри блоков if, for, elif, else
- Добавлены пустые строки в конце файлов
- Все двойные кавычки заменены на одинарные
- Сокращено количество дублирующегося кода в функции main файла
  encryptor.py
